clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusClusterProvider.groovy

```diff
@@ -220,19 +238,33 @@ class TitusClusterProvider implements ClusterProvider<TitusCluster>, ServerGroup
       cluster.serverGroups = clusterDataEntry.relationships[SERVER_GROUPS.ns]?.findResults { serverGroups.get(it) }
       cluster
     }
-    clusters
+    return clusters
   }
 
   /**
    * Translate server groups
    */
   private Map<String, TitusServerGroup> translateServerGroups(Collection<CacheData> serverGroupData) {
-    Collection<CacheData> allInstances = resolveRelationshipDataForCollection(serverGroupData, INSTANCES.ns, RelationshipCacheFilter.none())
-    Map<String, TitusInstance> instances = translateInstances(allInstances)
+    Collection<CacheData> allInstances = resolveRelationshipDataForCollection(serverGroupData, INSTANCES.ns, RelationshipCacheFilter.include(SERVER_GROUPS.ns))
+
+    def tasksByJobId = allInstances.groupBy { cacheData ->
+      cacheData.getAttributes().jobId
+    }.collectEntries { key, val ->
+      [(key): val.collect { cacheData ->
+        objectMapper.convertValue(cacheData.attributes.task, Task)
+      }]
+    }
+
+    Map<String, TitusInstance> instances = translateInstances(allInstances, serverGroupData)
 
     Map<String, TitusServerGroup> serverGroups = serverGroupData.collectEntries { serverGroupEntry ->
       String json = objectMapper.writeValueAsString(serverGroupEntry.attributes.job)
       Job job = objectMapper.readValue(json, Job)
+      if (job.tasks == null || job.tasks.isEmpty()) {
```

Comment:
not sure why this is needed

Reply:
we seem to be resolving instances from relationships later on, if this is not needed it might save us a call to get all task ids when caching the job

Reply:
You're right, this is not needed. We still need to call task ids though to correctly populate the relationships I think